### DISCUSSION TITLE
Problem uploading files

### DISCUSSION THREAD

**Discussion by elhennig** [abraunegg/onedrive#2732] [General]
I use the latest version from git and since yesterday  I get the following errors when trying to upload files to the private onedrive (Sharepoint still works for me):
```
ERROR: OneDrive returned an error with the following message:
  Error Message: Stream error in the HTTP/2 framing layer on handle 7F88D80E7AA0
  Calling Function: performHTTPOperation()
```
Is there a known issue with onedrive itself? The client has not changed in the meantime.
Any idea how to solve this? resync did not help

**Comment by abraunegg** [comment]
Please search this topic in issues and discussions - this has been explained a number of times.

**Reply by elhennig** [reply]
I remember, now. But why did it work some few days without having the option _force_http_11 = "true" set? Strange!

**Reply by abraunegg** [reply]
You need to ask:
* your linux distribution suppprt
* your isp

Something in the network path between your OS to Microsoft OneDrive does not support or has problems with http/2

**Reply by elhennig** [reply]
As it happens in my home office as well as in the office, the ISP is most likely not the problem. So it could only be the OS, opensuse leap 15.5. How can I check if the SO is supporting HTTP/2 for client connections?

I tried to remove  the option _force_http_11 = "true" some days ago and tried to run onedrive afterwards. That worked without problem until yesterday. And it only happens with the personal onedrive, not with sharepoint. I re-enabled the option and the config is working again. Thanks a lot for your hints!

**Reply by abraunegg** [reply]
So the issue comes down to your distribution.

Something in what they 'package' is what is causing the problem. It could be:
* Kernel related
* Network card driver related (fixed and/or wireless)
* Curl version related

Now .. from your other question for the exact same issue (https://github.com/abraunegg/onedrive/discussions/2698) - the version of curl your distribution is providing you with is from Mar 20 2023.

So .. are there HTTP/2 bugs in that version .. something for you to research - not me.

**Actions for you:**
* Submit a support request with your distribution
* Research HTTP/2 bugs in your version of libcurl - https://curl.se/docs/releases.html

**Reply by elhennig** [reply]
After upgrading to libcurl 8.8. I get the following error with the personal onedrive:
```
New items to upload to OneDrive: 10 .std.json.JSONException@std/json.d(641): Key not found: quickXorHash
----------------
??:? pure @safe noreturn std.exception.bailOut!(std.json.JSONException).bailOut(immutable(char)[], ulong, scope const(char)[]) [0x6d78ff]
??:? pure @safe inout(std.json.JSONValue)* std.exception.enforce!(std.json.JSONException).enforce!(inout(std.json.JSONValue)*).enforce(inout(std.json.JSONValue)*, lazy const(char)[], immutable(char)[], ulong) [0x84b444]
??:? inout pure ref @safe inout(std.json.JSONValue) std.json.JSONValue.opIndex(return immutable(char)[]) [0x7ef81d]
??:? bool syncEngine.SyncEngine.performUploadIntegrityValidationChecks(std.json.JSONValue, immutable(char)[], ulong, bool) [0x784098]
??:? void syncEngine.SyncEngine.uploadNewFile(immutable(char)[]) [0x775ecb]
??:? int syncEngine.SyncEngine.uploadNewLocalFileItemsInParallel(immutable(char)[][]).__foreachbody2(ulong, ref immutable(char)[]) [0x775166]
??:? void std.parallelism.ParallelForeach!(immutable(char)[][]).ParallelForeach.opApply(scope int delegate(ulong, ref immutable(char)[])).doIt() [0x6dce7e]
??:? void std.parallelism.run!(void delegate()).run(void delegate()) [0x7f2827]
??:? void std.parallelism.Task!(std.parallelism.run, void delegate()).Task.impl(void*) [0x7f230b]
??:? void std.parallelism.AbstractTask.job() [0x84c5a2]
??:? void std.parallelism.submitAndExecute(std.parallelism.TaskPool, void delegate()) [0x7f1f17]
??:? int std.parallelism.ParallelForeach!(immutable(char)[][]).ParallelForeach.opApply(scope int delegate(ulong, ref immutable(char)[])) [0x6dcd83]
??:? void syncEngine.SyncEngine.uploadNewLocalFileItemsInParallel(immutable(char)[][]) [0x775000]
??:? void syncEngine.SyncEngine.uploadNewLocalFileItems() [0x774f27]
??:? void syncEngine.SyncEngine.processNewLocalItemsToUpload() [0x76fef0]
??:? void syncEngine.SyncEngine.scanLocalFilesystemPathForNewData(immutable(char)[]) [0x76f4e5]
??:? void main.performStandardSyncProcess(immutable(char)[], monitor.Monitor) [0x5fe55e]
??:? _Dmain [0x5faffc]
```

Could this be related to the libcurl upgrade?

**Reply by abraunegg** [reply]
@elhennig 
> After upgrading to libcurl 8.8. I get the following error with the personal onedrive:

This looks like you are running one of the v2.5.x versions that has not been released. Please raise a new bug report providing all details including emailing a debug log which will be needed for resolution.

**Reply by elhennig** [reply]
I am using the build from scratch. Anyway, I found out that this affected only synchronisation with the --local-first option. I then did a --resync and afterwards the normal sync with or without --local-first works again.
Also, it seems to work without  the option _force_http_11 = "true".

**Reply by abraunegg** [reply]
@elhennig 
Not correct.

The *actual* issue stems from an actual code bug - which gets triggered in the following scenario:

* local file is being treated as new
* file technically exists online
* the API query online responds with valid JSON data but that JSON data is MISSING the [file] data points despite being requested. This should be treated as an API bug
* the application crashes because it is expecting the data in the response and not checking that it is there before use.

This has nothing to do with local first or forcing http 1.1 use

**Reply by elhennig** [reply]
I did not say it was due to local first or forcing http 1.1 use. The problem occurred after upgrading libcurl when using "local first"  and vanished after a "resync".
